cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/MetadataDocument.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.metadata.elastic;
+
+import co.cask.cdap.api.metadata.MetadataEntity;
+import co.cask.cdap.api.metadata.MetadataScope;
+import co.cask.cdap.spi.metadata.Metadata;
+import co.cask.cdap.spi.metadata.ScopedName;
+import org.elasticsearch.common.Strings;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * The document format that is indexed in Elastic.
+ */
+public class MetadataDocument {
+
+  /**
+   * Mapping for creating the ElasticSearch index. Although this is a property
+   * of the Elasticsearch index, it is defined here, because it maps 1:1 to the
+   * fields of this document class.
+   */
+  static final String MAPPING = ("{" +
```

Comment:
Instead of defining it as string concatenation in code, it would be better to put it in a resource file under `src/main/resources/metadata.mapping.json` and then load it with 

```
String MAPPING = Resources.toString(
  MetadataDocument.class.getClassLoader().getResource("metadata.mapping.json")
);
```

This helps readability and future modification.

Reply:
I did want to keep it with this this class, because it corresponds one:one to the members of the class, and any changes require change to the mapping, too. 
But I can make this change.

Reply:
done.